### PR TITLE
Update defaults for WandbLogger's run name and project name

### LINKED ISSUE #14054: WandbLogger : use random name instead of project as default name
## Proposed refactor
Use the wandb behavior (creating a random name) if no name is passed for the run or just let it `None` and wandb will deal with creating the name. 

### Motivation
I got 20 runs with all the same name which is not easy to navigate through 

### Pitch
Use a random name or name with a date or anything that would distinguish two runs. 

### Additional context
It's here: https://github.com/Lightning-AI/lightning/blob/9d02ad761c6c9360cba96e339d340c0cc2e572a8/src/pytorch_lightning/loggers/wandb.py#L300

cc @tchaton @rohitgr7 @awaelchli @morganmcg1 @borisdayma @scottire @manangoel99

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14054

Script to test the behavior if you want to play with it:
```py
import os

import torch
from torch.utils.data import DataLoader, Dataset

from pytorch_lightning import LightningModule, Trainer
from pytorch_lightning.loggers import WandbLogger


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def on_train_start(self) -> None:
        print(self.trainer.logger.save_dir)
        print(self.trainer.logger.name)
        print(self.trainer.logger.version)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    logger = WandbLogger(project="this")
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=1,
        limit_val_batches=1,
        limit_test_batches=1,
        num_sanity_val_steps=0,
        max_epochs=1,
        enable_model_summary=False,
        logger=logger,
    )
    trainer.fit(model, train_dataloaders=train_data)


if __name__ == "__main__":
    run()

```

Folder structure:
image

Wandb online record:
image

### Does your PR introduce any breaking changes? If yes, please list them.

Yes, but they are necessary to satisfy the users. 
- name: used to return the wandb.Run.name, but now returns the project name.
- Since name can't be None, we had to use a default name for the project name, and setting it to "lightning_logs" makes it more consistent with the other loggers. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @tchaton @rohitgr7 @borda @justusschock @awaelchli @morganmcg1 @borisdayma @scottire @manangoel99 @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/loggers/wandb.py [hunk 1]
```diff
@@ -450,13 +451,13 @@ def save_dir(self) -> Optional[str]:
 
     @property
     def name(self) -> Optional[str]:
```
Comment (rohitgr7) [inline src/pytorch_lightning/loggers/wandb.py:453]: I think it's a little off now since `name` argument and property are different. Instead, should we add/rename `project_name`?

**Review by rohitgr7 — APPROVED** [review]

File: src/pytorch_lightning/loggers/wandb.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/loggers/wandb.py:453]: I can't fix it in another way. The name here is tied to what is being required by the checkpointing logic, as explained in #14188. 

The only other way would be to hack an "isinstance check" in the ModelCheckpoint class and change the behavior for just wandb, but considering that #14188 is looking promising, I want to avoid this at all cost. 

Besides, it is also the same for the default logger, TensorBorad, where name does not represent the individual run's name, but the name of the entire "project'.

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/loggers/wandb.py (hunk 1 above)
Comment (rohitgr7) [reply src/pytorch_lightning/loggers/wandb.py:453]: #14188 looks good, but considering the logger individually, this is a little inconsistent since we have both `name` and `project_name` as arguments. I think we should have both these properties,  or which ever is required after #14188.

> Besides, it is also the same for the default logger, TensorBorad, where name does not represent the individual run's name, but the name of the entire "project'.

a wandb user might not care about other loggers at all.

Comment (awaelchli) [reply src/pytorch_lightning/loggers/wandb.py:453]: There will always be terminology in the 3rd party apis that does not mean the same thing in PL. The name property is on the base interface and I can't change it in just one logger. The discussion needs to be offloaded, I cannot address this here. If you want to add a project name property to the WandbLogger, feel free to do that. But at the moment, the implementation of "name" can't change, not until we have https://github.com/Lightning-AI/lightning/issues/14188

Comment (rohitgr7) [reply src/pytorch_lightning/loggers/wandb.py:453]: will create an issue to discuss this.

**Review by justusschock — APPROVED** [review]

**Comment by awaelchli** [comment]
gatekeeper rerun
